notification-server/src/bin/service.rs

```diff
@@ -414,7 +453,12 @@ async fn main() -> anyhow::Result<()> {
         .await
         .context("Failed to get processed block height")?
     {
-        height
+        catch_up_to_limit(
```

Comment:
Please add comments here so readers understand what is about to happen in the function and why.

Reply:
If you think the name of the function could be better, could you give an example of a name you would find better suited?